packages/jsxstyle-utils/src/getStyleCache.ts

```diff
@@ -14,17 +28,17 @@ function alreadyInjected() {
   );
 }
 
-function getStringHash(key) {
+function getStringHash(key: string, props: {}) {
```

Comment:
You're not using `props` here.  I take it this is because you have call sites that pass two params.

The 'correct' way to express this is probably something like:

```ts
export type GetClassNameForKey = (key: string, props: {}) => string;

let getClassNameForKey = getStringHash as GetClassNameForKey;
```

but if you wanted something quicker and more concise, you could make `props` optional:

```typescript
function getStringHash(key: string, props?: {}): string {
```

Reply:
Yeah, I didn’t use an optional param because it didn’t seem to change anything (probably because I haven’t hooked up tslint yet). @steadicat mentioned function overloads earlier (https://github.com/smyte/jsxstyle/pull/89#discussion_r151855318). All these options seem kinda gross, but function overloads seem the least gross right now.

Reply:
Personally, I think having a `GetClassNameForKey` type is the most semantic and least gross.  It's a bit more boilerplate, but it expresses your intent clearly (you have a callback that passes both arguments, you want to set it to one that only needs one argument, and you use casting to tell TS that the arity mismatch is OK.)